Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedGestureHand.cs

```diff
@@ -148,40 +149,47 @@ protected override void UpdateInteractions(SimulatedHandData handData)
                             {
                                 InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
 
-                                // Stop gestures
-                                CompleteHoldGesture();
-                                CompleteManipulationNavigationGesture();
+                                // Stop active gestures
+                                CompleteHold();
+                                CompleteManipulation();
+                                CompleteNavigation();
                             }
                         }
                         else if (Interactions[i].BoolData)
                         {
-                            if (!manipulationInProgress)
+                            if (manipulationInProgress)
+                            {
+                                UpdateManipulation();
+                            }
+
+                            if (navigationInProgress)
+                            {
+                                UpdateNavigation();
+                            }
+                            else
                             {
                                 if (cumulativeDelta.magnitude > manipulationStartThreshold)
```

Comment:
> if (cumulativeDelta.magnitude > manipulationStartThreshold) [](start = 31, length = 60)

Do we expect that there should ever be a case where you don't get manipulated started events, just hold start -> hold end -> navigation start -> navigation update -> navigation complete?

I'm curious because assuming there's only a hold going on and no navigation, it looks like it's possible that this would start navigation, and then on the next interaction update because a navigation is in progress, it will never start the manipulation. Or is it the case that, cumulativeDelta is only ever updated AFTER a hold has already been processed?

Reply:
Yeah, trying this out (pulling this down locally) I have to hold for a fairly significant amount of time to trigger the manipulation event - if I don't wait a long time, it only ever goes into navigation mode.

---
In reply to: [296315026](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5002#discussion_r296315026) [](ancestors = 296315026)

Reply:
I've changed it now so that the manipulation event starts as soon as you pinch.

Note that there will never be "canceled" events for `Manipulation` and `Navigation`, only for `Hold` when you start moving after `Hold` has already started. If i cancel the manipulation on `Hold` start then it will only work by moving immediately, which is probably not desirable either. 

With the current behavior the events are largely independent from each other, with the exception of canceling `Hold` when moving, and i think this is the least complicated way to do it.